### ISSUE TITLE
In 1-org + log_sinks.tf I had an issue with 'call to function "list" failed"

### ISSUE THREAD

**Issue by aitchpea** [terraform-google-modules/terraform-example-foundation#508]
Summary
* trying to find the approved way of overcoming issue with log_sinks.tf (or what I am doing that is 'bad')
* trying to understand the updating approach you have for modules (as updating the module is one solution for me)

I had an error when trying to perform terraform apply in "1-org" and there might be multiple options for solving (or I might be doing something 'wrong' - so apols).

File https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/1-org/envs/shared/log_sinks.tf

I got the error:

│ Error: Error in function call
│
│   on .terraform/modules/log_export_to_biqquery/main.tf line 33, in locals:
│   33:   bigquery_options = var.bigquery_options == null ? [] : var.unique_writer_identity == true ? list(var.bigquery_options) : []
│     ├────────────────
│     │ var.bigquery_options is object with 1 attribute "use_partitioned_tables"
│
│ Call to function "list" failed: the "list" function was deprecated in Terraform v0.12 and is no longer available; use tolist([ ... ]) syntax to write a literal list.

The above is due to the following block in log_sinks.tf:

![image](https://user-images.githubusercontent.com/83715131/126679133-4b96abf2-a9a8-4d58-9921-3f279fc69f7f.png)

So clearly there is the easy fix in syntax to make it work so I wondered if there was a better version of log_export_to_bigquery and was surprised to see there were so many major updates to the module (5.2 / 6.0 and even 7.0.0) whilst this code is  "~> 5.1.0".

Note: I updated my local copy of log_sinks.tf to use "~> 7.0.0" and it overcame the issue (but I have not seen all the code work to know that everything is correct - yet)

Hence my question on what the update policy is as my understanding is that this code is to be the Google opinionated template for how a company could bootstrap their GCP and so I would have envisaged a better tracking to the newest advocated modules.  I was wishing to use the code as a template - but I have concerns after seeing it lag so far behind the latest - and wonder what critical updates it is missing).

Views? Should I move to using the old version of Terraform (but versions.tf does say ">= 0.13" ) ? Am I doing something stupid (quite possible) ?

**Comment by bharathkkb** [comment]
> Call to function "list" failed: the "list" function was deprecated in Terraform v0.12

Were you running this locally with TF with 0.15+? The bootstrapped TF pipeline uses a 0.13 image and list() was [deprecated](https://www.terraform.io/upgrade-guides/0-15.html#legacy-configuration-language-features) in 0.15.


> updating approach you have for modules 

Thanks for the feedback. We will soon be implementing an autoupdate mechanism for automatically keeping upstream modules in sync with latest releases. In the meantime, happy to review a PR too to upgrade the log module.

**Comment by aitchpea** [comment]
I was running with TF 1.0.0 (which comes back to my question of what is the upgrade approach as I am loathe to downgrade my TF when designing a new org).

What do you recommend for my approach on patching - e.g. if I made the code >0.15 compatible it is not clear what versions you are publicly stating should be used (e.g. Corporates are often terrible at keeping up to date so if I upgraded to > 0.15 then are you alienating them?)

I would love to help patch it - and will do so when I understand how best to maintain the versioning you advocate

**Comment by bharathkkb** [comment]
We aim to keep it 0.13+ compatible and have been upgrading many upstream modules for 1.0 compat. For the most part changes in 1.0 are still compatible with 0.13 so unless we break something in 0.13, we can continue to keep minimum version as >0.13. We will soon start switching our testing images to 1.0 so that should let us flush out any other issues.

**Comment by pharveybattelle** [comment]
We have found the same issue, that envs/shared/log_sinks.tf and envs/shared/org_policy.tf are both pinned to versions of modules that don't work in newer versions of terraform.   I got past this particular error by unpinning the old module version, but am now striking a new error:  
``Step #1 - "tf plan validate all": Error: converting tfplan to CAI assets: adding resource changes to converter: adding resource create or update fetching asset: Error retrieving IAM policy for organization "XXXXXXXXX": googleapi: Error 403: The caller does not have permission, forbidden``
Note our org ID is correct in the error, I just removed it from the error message.
@aitchpea were you able to get 1-org code working with a newer version of Terraform?  We are currently using 1.0.4.
